app/src/main/java/org/kontalk/ui/view/ComposerBar.java

```diff
@@ -201,6 +207,14 @@ public void afterTextChanged(Editable s) {
 
                 if (mListener != null)
                     mListener.textChanged(s);
+
+                // covert ascii to emojis if preference set
+                boolean editing =false;
+                if (!editing && Preferences.getEmojiConverter(mContext)) {
+                    editing = true;
+                    s = smileyConvert(s);
+                    editing = false;
+                }
```

Comment:
* `s` is still replaced with itself
* your attempt to avoid an infinite loop won't work because `editing` must be a global variable. Again: Have you tested your code!?